Library/Homebrew/cmd/update-bash.sh

```diff
@@ -0,0 +1,256 @@
+    git config remote.origin.url https://github.com/Homebrew/homebrew.git
+    git config remote.origin.fetch +refs/heads/*:refs/remotes/origin/*
+  fi
+
+  if git remote show origin -n | grep -q "mxcl/homebrew"
+  then
+    git remote set-url origin https://github.com/Homebrew/homebrew.git
+    git remote set-url --delete origin ".*mxcl\/homebrew.*"
+  fi
+}
+
+repo_var() {
+  echo "$@" |
+    sed -e "s|$HOMEBREW_PREFIX||g" \
+        -e 's|Library/Taps/||g' \
+        -e 's|[^a-z0-9]|_|g' |
+    tr "[:lower:]" "[:upper:]"
+}
+
+upstream_branch() {
+  local upstream_branch
+  upstream_branch="$(git symbolic-ref refs/remotes/origin/HEAD 2>/dev/null |
+                     sed -e 's|refs/remotes/origin/||' )"
+  [ -z "$upstream_branch" ] && upstream_branch="master"
+  echo "$upstream_branch"
+}
+
+read_current_revision() {
+  git rev-parse -q --verify HEAD
+}
+
+pop_stash() {
+  [ -z "$STASHED" ] && return
+  git stash pop $QUIET_ARGS
+  if [ -n "$HOMEBREW_VERBOSE" ]
+  then
+    echo "Restoring your stashed changes to $DIR:"
+    git status --short --untracked-files
+  fi
+  unset STASHED
+}
+
+pop_stash_message() {
+  [ -z "$STASHED" ] && return
+  echo "To restore the stashed changes to $DIR run:"
+  echo "  'cd $DIR && git stash pop'"
+  unset STASHED
+}
+
+reset_on_interrupt() {
+  [ -z "$INITIAL_BRANCH" ] || git checkout "$INITIAL_BRANCH"
+  git reset --hard "$INITIAL_REVISION" $QUIET_ARGS
+  if [ -n "$INITIAL_BRANCH" ]
+  then
+    pop_stash
+  else
+    pop_stash_message
+  fi
+}
+
+pull() {
+  cd "$DIR" || return
+  TAP_VAR=$(repo_var "$DIR")
+  unset STASHED
+
+  # The upstream repository's default branch may not be master;
+  # check refs/remotes/origin/HEAD to see what the default
+  # origin branch name is, and use that. If not set, fall back to "master".
+  INITIAL_BRANCH="$(git symbolic-ref --short HEAD 2>/dev/null)"
+  UPSTREAM_BRANCH="$(upstream_branch)"
+  export HOMEBREW_UPDATE_AFTER"$TAP_VAR"="$(git rev-parse "$UPSTREAM_BRANCH")"
+
+  if [ -n "$(git status --untracked-files=all --porcelain 2>/dev/null)" ]
+  then
+    if [ -n "$HOMEBREW_VERBOSE" ]
+    then
+      echo "Stashing uncommitted changes to $DIR."
+      git status --short --untracked-files=all
+    fi
+    git -c "user.email=brew-update@localhost" \
+        -c "user.name=brew update" \
+        stash save --include-untracked $QUIET_ARGS
+    git reset --hard $QUIET_ARGS
+    STASHED="1"
+  fi
+
+  # Used for testing purposes, e.g., for testing formula migration after
+  # renaming it in the currently checked-out branch. To test run
+  # "brew update --simulate-from-current-branch"
+  if [ -n "$HOMEBREW_SIMULATE_FROM_CURRENT_BRANCH" ]
+  then
+    INITIAL_REVISION="$(git rev-parse -q --verify "$(upstream_branch)")"
+    CURRENT_REVISION="$(read_current_revision)"
+    if ! git merge-base --is-ancestor "$INITIAL_REVISION" "$CURRENT_REVISION"
+    then
+      echo "Your HEAD is not a descendant of $UPSTREAM_BRANCH!" >&2
+      exit 1
+    fi
+    return
+  fi
```

Comment:
The `--simulate-from-current-branch` feature is completely broken because `$CURRENT_REVISION` is never passed on to the command responsible for the reporting. (Also, `git fetch` will still run in this mode, though it doesn't need to – and probably shouldn't for efficiency reasons.)